typewise-alert.cpp

```diff
@@ -11,61 +17,80 @@ BreachType inferBreach(double value, double lowerLimit, double upperLimit) {
   return NORMAL;
 }
 
-BreachType classifyTemperatureBreach(
-    CoolingType coolingType, double temperatureInC) {
-  int lowerLimit = 0;
-  int upperLimit = 0;
-  switch(coolingType) {
-    case PASSIVE_COOLING:
-      lowerLimit = 0;
-      upperLimit = 35;
-      break;
-    case HI_ACTIVE_COOLING:
-      lowerLimit = 0;
-      upperLimit = 45;
-      break;
-    case MED_ACTIVE_COOLING:
-      lowerLimit = 0;
-      upperLimit = 40;
-      break;
+std::string controllerAlert::buildAlertMessageString()
+{
+  std::stringstream ssHeader;
+  std::stringstream ssbreachType;
+  ssHeader << std::hex << header;
+  ssbreachType << std::hex << breachType;
+  std::string controllerMessage = ssHeader.str() +" : " + ssbreachType.str() + "\n";
+  return controllerMessage;
+}
+
+std::string EmailAlert::buildAlertMessageString()
+{
+  std::string eMailMessage = "";
+  if(breachType == TOO_LOW){
+    eMailMessage = "To: " + eMail + "\n" + "Hi, the temperature is too low\n";
   }
-  return inferBreach(temperatureInC, lowerLimit, upperLimit);
+  if(breachType == TOO_HIGH){
+    eMailMessage = "To: " + eMail + "\n" + "Hi, the temperature is too high\n";
+  }
+  return eMailMessage;
 }
 
-void checkAndAlert(
-    AlertTarget alertTarget, BatteryCharacter batteryChar, double temperatureInC) {
+bool AlertHandler::sendAlertToTarget(std::string alertMessage){
+  bool isSuccess = true;
+  std::cout << alertMessage;
+  return isSuccess;
+}
 
-  BreachType breachType = classifyTemperatureBreach(
-    batteryChar.coolingType, temperatureInC
-  );
+temperatureLimitsPair AlertHandler::getTemperatureLimits(CoolingType coolingType)
+{
+  return tempLimitsMap.at(coolingType);
+}
 
-  switch(alertTarget) {
-    case TO_CONTROLLER:
-      sendToController(breachType);
-      break;
-    case TO_EMAIL:
-      sendToEmail(breachType);
-      break;
-  }
+BreachType AlertHandler::getTemperatureBreach(CoolingType coolingType, double temperatureInC)
+{
+  std::pair<double, double> temperatureLimits = getTemperatureLimits(coolingType);
+  double lowerLimit = temperatureLimits.first;
+  double upperLimit = temperatureLimits.second;
+  return inferBreach(temperatureInC, lowerLimit, upperLimit);
 }
 
-void sendToController(BreachType breachType) {
-  const unsigned short header = 0xfeed;
-  printf("%x : %x\n", header, breachType);
+bool AlertHandler::createAlertTarget(AlertTarget alertTarget, BreachType breachType)
```

Comment:
This function is a factory which creates test stubs for controller and email (classes for the purpose of testing, not real functionality). Try separating it from this class and moving it along with the test code to test-alerts.cpp.